### PR TITLE
Add option to enable tech preview features

### PR DESCRIPTION
**Pull request by danifr**
https://github.com/treydock/puppet-module-keycloak/issues/94

Probably worth to add some tests for this too.
Not sure if I should also support a way to disable/enable per feature.

### CODE REVIEW

File: manifests/config.pp [hunk 1]
```diff
@@ -37,6 +37,17 @@
...
 
+  if $keycloak::tech_preview_features {
+    file { "${keycloak::install_base}/standalone/configuration/profile.properties":
+      ensure  => 'file',
+      owner   => $keycloak::user,
+      group   => $keycloak::group,
+      content => template('keycloak/profile.properties.erb'),
+      mode    => '0750',
+      require => File["${keycloak::install_base}/standalone/configuration"],
+    }
```
Comment (treydock) [inline manifests/config.pp]: Would be good to `notify` the Keycloak service class, `Class['keycloak::service']`. Also to allow undoing this change maybe set `ensure` to `$profile_properties_ensure` and in `init.pp` set the value based on presence of `tech_preview_features`.  This way if someone no longer set `tech_preview_features` the file will get deleted and Keycloak restarted.

File: manifests/init.pp [hunk 2]
```diff
@@ -250,12 +252,19 @@
   Boolean $restart_sssd = true,
   Optional[Stdlib::Absolutepath] $service_environment_file = undef,
   Enum['standalone', 'clustered'] $operating_mode = 'standalone',
+  Optional[Array] $tech_preview_features = undef,
```
Comment (treydock) [inline manifests/init.pp]: Make this just `Array` with empty array as default.

File: manifests/init.pp [hunk 3]
```diff
@@ -250,12 +252,19 @@
...
 ) {
 
   if ! $facts['os']['family'] in ['RedHat','Debian'] {
     fail("Unsupported osfamily: ${facts['os']['family']}, module ${module_name} only support osfamilies Debian and Redhat")
   }
 
+  $profile_properties_ensure = $tech_preview_features ? {
```
Comment (treydock) [inline manifests/init.pp]: If this becomes just empty array can do something like `if empty($tech_preview_features) { `

File: templates/profile.properties.erb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<% if scope['keycloak::tech_preview_features'] -%>
```
Comment (treydock) [inline templates/profile.properties.erb]: Once `tech_preview_features` is only an array, can remove this if statement.

**Comment by danifr** [comment]
@treydock sorry I've been busy and I couldn't address your comments...
Thanks for taking the over!

**Comment by treydock** [comment]
This was released in v5.8.0

**Pull request merged**